### PR TITLE
fix: Clear the cache - git limit number of repos

### PR DESCRIPTION
**Pull request by AnaghHegde**
## Description

> Clear the number of git connected repos for the orgId_InstanceId on the connect API call.

Fixes # (issue)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Locally

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java [hunk 1]
```diff
@@ -129,6 +127,7 @@
     INVALID_JS_ACTION(400, 4040, "Something went wrong while trying to parse this action. Please check the JS object for errors.", AppsmithErrorAction.DEFAULT, null, ErrorType.BAD_REQUEST),
     CYCLICAL_DEPENDENCY_ERROR(400, 4041, "Cyclical dependency error encountered while parsing relationship [{0}] where the relationship is denoted as (source : target).", AppsmithErrorAction.DEFAULT, "Cyclical Dependency in Page Load Actions", ErrorType.CONFIGURATION_ERROR),
     CLOUD_SERVICES_ERROR(500, 5012, "Received error from cloud services {0}", AppsmithErrorAction.DEFAULT, null, ErrorType.INTERNAL_ERROR),
+    GIT_APPLICATION_LIMIT_ERROR(400, 5103, "You have reached the maximum number of private git repo counts which can be  connected to the organization. Please reach out to Appsmith support to opt for commercial plan.", AppsmithErrorAction.DEFAULT, null, ErrorType.EE_FEATURE_ERROR),
```
Comment (abhvsn) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java:130]: Why are we not following the nomenclature as per earlier error basically all `4xx` httpErrorCode have `4xxx` app error code?

Comment (AnaghHegde) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/exceptions/AppsmithError.java:130]: Nice catch. I will update this.

**Review by abhvsn — APPROVED** [review]
LGTM

Are we going to add TC as a separate PR?

**Comment by AnaghHegde** [comment]
@abhvsn Tests will be a separate PR.

**Review by abhvsn — APPROVED** [review]

**Review by abhvsn — APPROVED** [review]

**Pull request merged**